opsgenie.yaml

```diff
@@ -84,6 +84,142 @@ systems:
                       call_function: opsgenie.snooze
                       #  alert_Id is exported from the event
 
+      schedules:
+        driver: web
+        rawAction: request
+        parameters:
+          URL: https://api.opsgenie.com/v2/schedules{{ with .ctx.scheduleId }}/{{ . }}?identifierType={{ default "name" $.ctx.scheduleIdType }}{{ end }}
+          method: GET
+          header:
+            Authorization: GenieKey {{ .sysData.API_KEY }}
+          form:
+            limit: "100"
+        export:
+          schedule: $data.json.data
+          schedules: $data.json.data
+
+        meta:
+          description:
+            - This function list all on-call schedules or fetch a schedule detail if given a schedule identifier.
+            - highlight: This function only fetches first 100 schedules when listing.
+              type: important
+
+          inputs:
+            - name: scheduleId
+              description: The name or ID or the schedule of interest; if missing, list all schedules.
+            - name: scheduleIdType
+              description: The type of the identifier, :code:`name` or :code:`id`.
+          exports:
+            - name: schedule
+              description: For fetching detail, the data structure that contains the schedule detail
+            - name: schedules
+              description: For listing, a list of data structure contains the schedule details
+
+          notes:
+            - See below for example
+            - example: |
+                ---
+                workflows:
+                  steps:
+                    - call_function: opsgenie.schedules
+
+      who-is-oncall:
```

Comment:
maybe remove the dashes to keep it consistent with their command?

Reply:
done